src/transform/with_lengthscale.jl

```diff
@@ -0,0 +1,34 @@
+"""
+    with_lengthscale(kernel::Kernel, lengthscale::Real)
+    with_lengthscale(kernel::Kernel, lengthscales::AbstractVector{<:Real})
+
+Constructs a transformed kernel with `lengthscale`. If `lengthscales` is passed
+as a vector instead, constructs an "ARD" kernel with different lengthscales for
+each dimension.
```

Comment:
```suggestion
Constructs a transformed kernel with `lengthscale`.
If a vector `lengthscales` is passed instead, constructs an "ARD" kernel with different lengthscales for each dimension.
```
Minor grammar thing -- I think this way makes more sense to me.